reducers/plugins/index.js

```diff
@@ -5,6 +5,44 @@ import {combineReducers} from 'redux';
 
 import {ActionTypes} from 'utils/constants.jsx';
 
+function addMenuItem(menu, data) {
+    if (menu.id === data.parentId) {
+        menu.subMenu = [...menu.subMenu, data];
+    } else if (menu.subMenu) {
+        menu.subMenu = menu.subMenu.map((item) => {
+            const subMenu = {...item};
+            return addMenuItem(subMenu, data);
+        });
+    }
+    return menu;
+}
+
+function buildSubMenu(state, action) {
+    const nextState = {...state};
+    const currentArray = nextState[action.name];
+    if (!currentArray) {
+        return action;
+    }
+    const root = currentArray.find((c) => c.subMenu);
```

Comment:
hmm, would this work if you had more than 1 plugin registered submenus? don't we have to find all the menus for _this_ `pluginId` and for those find the ones we need to update?

Reply:
great catch, thank you.  updated to `const root = currentArray.find((c) => c.subMenu && c.pluginId === data.pluginId);`

Reply:
@scottleedavis don't you to filter here, since the same plugin could've registered multiple menu/submenus?